src/instrument.js

```diff
@@ -107,8 +126,42 @@ async function findInstruments(oreAccountName, activeOnly = true, category = und
   return instruments;
 }
 
+function sortInstruments(instruments, rightName, sortOrder = "cheapestThenMostRecent") {
+  // sorts the instruments depending on the search criteria : 
+  // cheapestThenMostRecent - returns the cheapest instrument for the right and if there are more than one with the same price, then returns the latest created/updated instrument
+  // mostRecent - returns the latest instrument created/updated for the right
+  let sortedInstruments;
+  let cheapestInstrument;
+  let cheapestPrice;
+  let cheapestInstruments;
+
+  const sortTypes = {
+    1: "cheapestThenMostRecent",
+    2: "mostRecent"
+  }
+
+  switch (sortOrder) {
+    case sortTypes[1]:
+      sortedInstruments = sortByCheapestRight.bind(this)(instruments, rightName)
+      cheapestInstrument = sortedInstruments[0]
+      cheapestPrice = this.getRight(cheapestInstrument, rightName).price_in_cpu;
+      cheapestInstruments = sortedInstruments.filter(instrument => this.getRight(instrument, rightName).price_in_cpu === cheapestPrice)
+
+      // if more than one instrument has the same price for the right, get the most recently updated
+      if (cheapestInstruments.length > 0) {
```

Comment:
If `sortByMintedAt` returns the entire sorted array, we don't need this qualifier

Reply:
I am not sure I understand what you are trying to say.

Reply:
ok! got it